File: stdin_primer/ruby/05-02_integer_space_step2.rb
# 2 つの整数の半角スペース区切りの入力 (paizaランク D 相当)
# https://paiza.jp/works/mondai/stdin_primer/stdin_primer__integer_space_step2

# 入出力例
INPUT1 = <<~"EOS"
  8 13
EOS
OUTPUT1 = <<~"EOS"
  8
  13
EOS

INPUT2 = <<~"EOS"
  1234 5678
EOS
OUTPUT2 = <<~"EOS"
  1234
  5678
EOS

# 解答例
# 入力
a, b = gets.split.map(&:to_i)
# 出力
puts(a)
puts(b)

File: stdin_primer/haskell/05-02_integer_space_step2.hs
-- 2 つの整数の半角スペース区切りの入力 (paizaランク D 相当)
-- https://paiza.jp/works/mondai/stdin_primer/stdin_primer__integer_space_step2

toInt :: String -> Int
toInt x = read x :: Int

toStr :: Int -> String
toStr x = show x :: String

main :: IO ()
main = do
  inputStr <- getLine

  let strNums = words inputStr
  let intNums = map toInt strNums

  putStrLn (toStr (intNums !! 0))
  putStrLn (toStr (intNums !! 1))
